### ISSUE TITLE
extract_gradient! does not work for outputting to an Array

### ISSUE THREAD

**Issue by KristofferC** [JuliaDiff/ForwardDiff.jl#163]
While this is an internal function, I think this behavior is weird. It should just extract the partials into the passed vector I presume?

```jl
julia> ForwardDiff.extract_gradient!(zeros(3), ForwardDiff.Dual(1.0, (2.0,3.0,4.0)))
------ MethodError --------------------- Stacktrace (most recent call last)

 [1] — extract_gradient!(::Array{Float64,1}, ::ForwardDiff.Dual{3,Float64}) at gradient.jl:40
 [2] — copy!(::Array{Float64,1}, ::ForwardDiff.Partials{3,Float64}) at multidimensional.jl:614

MethodError: no method matching getindex(::Tuple{Float64,Float64,Float64}, ::CartesianIndex{1})
```

**Comment by KristofferC** [comment]
Maybe a Base issue if `getindex` should be defined on tuples for `CartesianIndex{1}`?

**Comment by KristofferC** [comment]
This happended here: https://travis-ci.org/JuliaDiff/ForwardDiff.jl/jobs/180792314#L356. strainge.

**Comment by jrevels** [comment]
Good catch; it's because I forgot to define `inearindexing` for `Partials` as `LinearFast`. Fixed in #162 (probably should've been a separate PR but I was lazy).
